pyproject.toml

```diff
@@ -34,20 +34,20 @@ eth-keyfile = "^0.8.1"
 docs = ["sphinx", "enum-tools", "furo"]
 
 [tool.poetry.group.dev.dependencies]
-pytest = "^7.2.2"
-black = "^23.1.0"
-poethepoet = "^0.24.4"
+pytest = "^8.2.2"
+black = "^24.4.2"
+poethepoet = "^0.27.0"
 coverage = "^7.2.1"
 pytest-asyncio = "^0.21.1"
```

Comment:
Note: Not updating `pytest-asyncio` as it breaks our fixtures. As per the information in the [release notes](https://github.com/pytest-dev/pytest-asyncio/releases/tag/v0.23.8) they suggest continuing to use version `0.21`